bsp/src/org/jetbrains/bsp/protocol/session/BspSession.scala

```diff
@@ -105,16 +111,20 @@ class BspSession private(bspIn: InputStream,
     }
 
     override def close(): Unit = {
-      writer.flush()
+      writer.close()
     }
   }
 
 
   private def bspTraceLogger: PrintWriter = {
-    val dirs = Paths.get(PathManager.getLogPath, "bsp")
-    Files.createDirectories(dirs)
-    val stamp = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH-mm-ss").format(LocalDateTime.now())
-    val logfile = new File(dirs.toFile, s"bsp-protocol-trace-$stamp.log")
+    val logfile = sys.env.get("BSP_TRACE_PATH")
```

Comment:
This helps to write all output to a single .log file, indexed in IDE, instead of fetching new files every time. The default behaviour however, will remain the same.